### PR TITLE
Prepare for PyPI releases via name change to `almanack` and automated release + publishing capabilities

### LINKED ISSUE #19: Explore PyPI releases for the book content

### LINKED ISSUE #41: Consider updating package name (and project) to `almanack`

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR readies the project for consistent publishing on PyPI as a package by updating the project name to `almanack` (there are many existing uses of the word `almanac` in Python packages + `almanack` harks to common older English spellings used with farmer almanac publications). Along with this I prepared release drafter and trusted publishing to PyPI through GitHub actions and an initial push to PyPI (trusted publishing cannot otherwise be configured for first-time pushes to my knowledge).

After this PR is merged, I plan to update the GitHub repository to be named "almanack", will publish the draft release which is created, and verify the PyPI package is uploaded as expected.

Thanks for any feedback you may have!

Closes #41 
Closes #19 
References #42 (I've requested an organization for community-based management of package resources but this may take time to realize)

## What is the nature of your change?

- [ ] Content additions or updates (adds or updates content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [x] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own contributions.
- [x] I have commented my content, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to related documentation (outside of book content).
- [x] My changes generate no new warnings.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests that prove my additions are effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: pyproject.toml
Comment (gwaybio) [inline pyproject.toml:3]: why is this bump down?

File: src/book/garden-circle/contributing.md
Comment (gwaybio) [inline src/book/garden-circle/contributing.md:142]:
```suggestion
We utilize [semantic versioning](https://en.wikipedia.org/wiki/Software_versioning#Semantic_versioning) ("semver") to distinguish between major, minor, and patch releases.
```

the phrase "in order to" is almost always fluff that can be easily removed :)

File: src/book/garden-circle/contributing.md
Comment (gwaybio) [inline src/book/garden-circle/contributing.md:144]: Maybe it is worth being a bit more specific than "related"?

File: src/book/garden-circle/contributing.md
Comment (gwaybio) [inline src/book/garden-circle/contributing.md:148]:
```suggestion
Publishing Software Gardening Almanack releases involves several manual and automated steps.
```

Active voice

File: src/book/garden-circle/contributing.md
Comment (gwaybio) [inline src/book/garden-circle/contributing.md:151]: This line feels a bit out of place given the previous "see below..." language

You immediately point to a different resource for someone to view? perhaps an easy fix would be to add this sentence later, after describing specs

File: src/book/garden-circle/contributing.md
Comment (gwaybio) [inline src/book/garden-circle/contributing.md:153]:
```suggestion
## Version specifications 

- [`poetry-dynamic-versioning`](https://github.com/mtkennerly/poetry-dynamic-versioning) leveraging [`dunamai`](https://github.com/mtkennerly/dunamai) creates version data based on [git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) and commits.
- [GitHub Releases](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases) automatically create git tags
- [`release-drafter`](https://github.com/release-drafter/release-drafter) infers and describes changes since last release
```

Please double check (and/or ignore) my edits here. I switched to active voice and tried to add a bit of structure for increasing clarity

File: src/book/garden-circle/contributing.md
Comment (gwaybio) [inline src/book/garden-circle/contributing.md:154]:
```suggestion
### Version specification process

```

File: src/book/garden-circle/contributing.md
Comment (gwaybio) [inline src/book/garden-circle/contributing.md:159]:
```suggestion
1. On publishing the release, another GitHub Actions workflow defined in `publish-pypi.yml` will automatically build and deploy the Python package to PyPI (utilizing the earlier modified `pyproject.toml` semantic version reference for labeling the release).
```

**Review by gwaybio — APPROVED** [review]
minor comments - LGTM and ready to merge once you're happy

File: pyproject.toml
Comment (d33bs) [reply pyproject.toml:3]: Thanks for the double check here, the earlier version entry was I believe prior to dynamic versioning capabilities and this is a correction towards that effect (where `0.0.0` is the preference via `"Poetry still requires the tool.poetry.version field to be present in pyproject.toml, but you are encouraged to use version = "0.0.0" as a standard placeholder."` from [here](https://github.com/mtkennerly/poetry-dynamic-versioning)).

File: src/book/garden-circle/contributing.md
Comment (d33bs) [reply src/book/garden-circle/contributing.md:142]: Thanks!

File: src/book/garden-circle/contributing.md
Comment (d33bs) [reply src/book/garden-circle/contributing.md:144]: Thanks good point, I'll add some specificity here.

File: src/book/garden-circle/contributing.md
Comment (d33bs) [reply src/book/garden-circle/contributing.md:151]: Thanks, great point - I'll move things around a bit to help with this.

File: src/book/garden-circle/contributing.md
Comment (d33bs) [reply src/book/garden-circle/contributing.md:153]: Thanks for this, looks a lot better!

File: src/book/garden-circle/contributing.md
Comment (d33bs) [reply src/book/garden-circle/contributing.md:159]: Thanks! A lot more clear.

File: src/book/garden-circle/contributing.md
Comment (d33bs) [reply src/book/garden-circle/contributing.md]: Updated in a further commit.

File: src/book/garden-circle/contributing.md
Comment (d33bs) [reply src/book/garden-circle/contributing.md]: Reorganized this with a further commit.

**Comment by d33bs** [comment]
Thanks @gwaybio for the review! Merging this in now.

**Comment by d33bs** [comment]
Note: we saw failures in the PyPI release process after merging this PR. I believe this is due to the fact that the repository had not yet been renamed to `almanack` prior to the GitHub release being published which as a result caused a failure in alignment with PyPI trusted publisher settings (which expect `almanack` in the naming of the repo). On retrying this after renaming the repo, GH Actions containers appear to re-use the original repo name when re-running them. I expect that this might change on subsequent releases and new actions jobs being created. We'll give this a try after the next PR merge.

**Pull request merged**
